frontend/src/metabase/dashboard/components/DashboardGrid.styled.tsx

```diff
@@ -6,10 +6,55 @@ interface DashboardCardProps {
 }
 
 export const DashboardCard = styled.div<DashboardCardProps>`
+  position: relative;
+  z-index: 2;
+
+  /**
+  * Dashcards are positioned absolutely so each one forms a new stacking context.
+  * The dashcard user is currently interacting with needs to be positioned above other dashcards
+  * to make sure it's not covered by absolutely positioned children of neighboring dashcards.
+  *
+  * @see https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_positioned_layout/Understanding_z-index/Stacking_context
+  */
+  &:hover {
```

Comment:
Should we add `:focus-within` here too? So when we navigate to the dashcard actions using keyboard it won't be covered by the card above.

It'd be great to add tests to cover this case too.

```suggestion
  &:hover, &:focus-within {
```

#### This PR
![Screenshot 2023-08-08 at 3 26 57 PM](https://github.com/metabase/metabase/assets/1937582/39174233-be72-467a-a74f-3b5d8396274f)

#### With `:focus-within`
![Screenshot 2023-08-08 at 3 27 19 PM](https://github.com/metabase/metabase/assets/1937582/f9508d2c-2d3a-485a-988a-c66767a70eef)